appserver/admingui/common/src/main/java/org/glassfish/admingui/common/util/RestUtil.java

```diff
@@ -893,11 +893,21 @@ public static RestResponse post(String address, Object payload, String contentTy
 
     public static RestResponse post(String address, Map<String, Object> payload) {
         WebTarget target = getJerseyClient().target(address);
-        // Map formData = buildMultivalueMap(payload); Commented for sakes of reference in future rather that removal
-        Response cr = target
+        Response cr;
+        
+        if(address.endsWith("system-properties")) {
+        cr = target
                 .request(RESPONSE_TYPE)
                 .cookie(new Cookie(REST_TOKEN_COOKIE, getRestToken()))
                 .post(Entity.entity(payload, MediaType.APPLICATION_JSON), Response.class);
+        } else {
+            MultivaluedMap formData = buildMultivalueMap(payload);
+            cr = target
+                .request(RESPONSE_TYPE)
+                .cookie(new Cookie(REST_TOKEN_COOKIE, getRestToken()))
+                .post(Entity.entity(formData, MediaType.APPLICATION_FORM_URLENCODED), Response.class);
+        }
```

Comment:
This else branch looks very similar to the first one. Why not inline condition check in `post` method's 1st argument (and then extract it to some method) and have `cr` initialized on definition?

Reply:
I think it would also be fine to do it that way, although given the if case is true there is no need to create the formData Map instance as it would never be used. Personally, from a readability point of view I feel it's nicer this way as comparatively the line would get much longer using an inline condition

Reply:
> I think it would also be fine to do it that way, although given the if case is true there is no need to create the formData Map instance as it would never be used.

With `?` `formData` would also be created only when needed.

> Personally, from a readability point of view I feel it's nicer this way as comparatively the line would get much longer using an inline condition

With method extracted to do that it would be more readable IMO.
Pseudo:
```
...
   val cr = target
                .request(RESPONSE_TYPE)
                .cookie(new Cookie(REST_TOKEN_COOKIE, getRestToken()))
                .post(getEntityForAddress(address, payload), Response.class);

   RestResponse rr = RestResponse.getRestResponse(cr);
   return rr;
}

getEntityForAddress(address, payload) {
    return address.condition() ? payload :  buildMultivalueMap(payload);
}
```

Then one could provide test for `getEntityForAddress` (which would be easier to test that decision making).

Reply:
Made that change now I know what you mean properly, much nicer way to do it, thanks!